src/storage.rs

```diff
@@ -593,82 +587,83 @@ impl Store {
             // We're locked, so can't update anything.
             return self.imports.clone();
-        }
-
-        // Determine which packages were used.
-        let mut used_packages = SortedMap::new();
-        for package in graph.nodes.iter() {
-            used_packages.insert(package.name, false);
-        }
-
-        // If we succeeded, also record which packages need fresh imports.
-        if let Conclusion::Success(success) = conclusion {
-            for &node_idx in &success.needed_fresh_imports {
-                let package = &graph.nodes[node_idx];
-                used_packages.insert(package.name, true);
-            }
-        }
+        };
 
         let mut new_imports = ImportsFile {
             audits: SortedMap::new(),
         };
-        for import_name in self.config.imports.keys() {
-            let live_audit_file = self
-                .imported_audits()
-                .get(import_name)
-                .expect("Live audits missing for import?");
-
-            // Is this the first time we're doing this import? If it is we'll
-            // want to effectively force-update this peer.
-            let first_import = !self.imports.audits.contains_key(import_name);
-
-            // Create the new audits file, and copy criteria into it on a
-            // per-package basis, only including used packages.
-            // We always use the live version of criteria in the new file.
+        for (import_index, (import_name, live_audits_file)) in
+            live_imports.audits.iter().enumerate()
+        {
             let mut new_audits_file = AuditsFile {
-                criteria: live_audit_file.criteria.clone(),
+                criteria: live_audits_file.criteria.clone(),
                 audits: SortedMap::new(),
             };
-            for (&package, &need_fresh_imports) in &used_packages {
-                // Always pull audit entries from the live version to pull in
-                // revocations, violations, updated notes, etc.
-                // Filter out entries which are fresh unless we're updating
-                // entries for this package or it's a violation.
-                let update_package = first_import || need_fresh_imports || force_update;
-                let audits = live_audit_file
+            for (pkgidx, result) in results.iter().enumerate() {
+                let package = &graph.nodes[pkgidx];
+
+                // Don't import audits for first-party packages.
+                if !package.is_third_party {
+                    continue;
+                }
+
+                // If the audit succeeded, get the set of required audits.
+                let required_edges = match result {
+                    Some(resolver::ResolveResult {
+                        required_edges: Some(required_edges),
+                        ..
+                    }) => Some(required_edges),
+                    _ => None,
+                };
```

Comment:
Yeah, I considered doing the shorter version, but I figured I'd just write out the match statement instead, because it's fairly clear what it's checking.